projects/fsastorefrontlib/src/occ/adapters/consent-management/occ-consent.adapter.ts

```diff
@@ -62,15 +66,27 @@ export class OccConsentAdapter implements ConsentAdapter {
       .pipe(catchError((error: any) => throwError(error.json())));
   }
 
-  getOBOCustomerList(userId: string): Observable<any> {
+  getOBOCustomerList(
+    userId: string,
+    searchConfig?: FSSearchConfig
+  ): Observable<OBOCustomerList> {
     const url = this.occEndpointService.buildUrl('oboConsentCustomers', {
       urlParams: {
         userId,
       },
     });
-    const params = new HttpParams({ fromString: FULL_PARAMS });
+    let params = new HttpParams({ fromString: FULL_PARAMS });
+    if (searchConfig) {
```

Comment:
Seems this first if is redundant since you are using chaining operator in the next if? Please check.

Reply:
I'll remove first if and add additional for `params = params.set('currentPage', searchConfig?.currentPage)`, so it won't be nested if anymore.

Reply:
And please set multiple params with `append` method instead like this. You can use `updateOBOPermission` in this file as a reference.

Reply:
You can leave it like this, but from official Angular [documentation](https://angular.io/api/common/http/HttpParams#description) it's clear that `append` is used for this purpose specifically, while `set` replaces the value for a parameter (but it's obviously also working). More on this dilemma can be found [here](https://stackoverflow.com/questions/45459532/why-httpparams-doesnt-work-in-multiple-line-in-angular-4-3/45459672#45459672)

Reply:
[Here](https://stackoverflow.com/questions/51980630/what-is-the-difference-between-the-set-and-append-methods-of-angulars-httpparams) through examples you can see the difference between `set` and `append`. In our case, I'll keep with `set`, since we want a unique value for a particular parameter key. Append for me has a more sense if we need more values for a particular parameter key.